walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,123 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    final Collection<String> requestAttendees = request.getAttendees();
+    final long requestDuration = request.getDuration();
+    ArrayList<TimeRange> eventList = createFlattenedEventList(events, requestAttendees);
+
+    if(requestDuration > TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
+    if(requestAttendees.isEmpty() || eventList.isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
```

Comment:
These are some nice things to check - I recommend putting these two if statements before line 28, since `createFlattenedEventList` is your expensive operation. If you can avoid needing to do it, because the time range is out of bounds or because there are no attendees/events, you might as well :)